src/printapp/static/js/app.js

```diff
@@ -58,6 +58,14 @@ App.ApplicationController = Ember.Controller.extend({
       controller.set('loadingFailed', true);
       if (data.status === 504) {
         controller.set('uniflowIsDown', true);
+      } 
+      else if (data.status === 401) {
+        //Log out by posting to the logout api, which deletes the httponly cookie keeping the user logged in.
+        var form = document.createElement('form');
+        document.body.appendChild(form);
+        form.method = 'POST';
+        form.action = '/api/logout';
+        form.submit();
```

Comment:
Could we factor these 5 sign out lines into a separate method? This will reduce code duplication.